lib/pbench/test/unit/server/auth/test_auth.py

```diff
@@ -20,6 +20,84 @@
 from pbench.test.unit.server.conftest import jwt_secret
 
 
+@pytest.fixture
+def mock_oidc(monkeypatch, on_disk_server_config):
+    def mock_connection(
+        client_id: str, public_key: Optional[str] = None
+    ) -> PbenchServerConfig:
+        """Create a mocked Connection object whose behavior is driven off of
+        the realm name and / or client ID.
+
+        NOTE: we build our own PbenchServerConfig here rather than using the
+        `server_config` fixture as that's `session` scope and will pollute
+        other tests with our custom `openid` configuration.
+
+        Args:
+            client_id : A client ID used to influence the behavior of the mocked
+                Connection object
+            public_key : Optional public key to use
+
+        Returns:
+            A PbenchServerConfig object for use constructing an
+                OpenIDClient object
+        """
+        server_url = "https://example.com"
+        realm_name = "camelot"
+        audience = "server"
+        secret = "shhh"
+        cfg_file = on_disk_server_config["cfg_dir"] / "pbench-server.cfg"
+        server_config = PbenchServerConfig(str(cfg_file))
+        config = server_config._conf
+        config["openid"] = {
+            "server_url": server_url,
+            "client": client_id,
+            "audience": audience,
+            "realm": realm_name,
+            "secret": secret,
+        }
+        public_key = "abcdefg" if public_key is None else public_key
+
+        class MockResponse:
+            """A mocked requests.Response object which just provides a json()
+            method.
+            """
+
+            def __init__(self, json: str):
+                self._json = json
+
+            def json(self):
+                return self._json
+
+        class MockConnection:
```

Comment:
Arguably, `MockResponse` should be defined inside `MockConnection` or even inside `MockConnection.get()`.